cli/src/main/scala/scala/scalanative/cli/ScalaNativeP.scala

```diff
@@ -32,24 +39,89 @@ object ScalaNativeP extends CaseApp[POptions] {
       System.err.println(s"Ignoring non existing path: $path")
     }
 
+    if (options.nirFiles) deserializeFiles(classpath, args.all)
+    else deserializeClassesOrObjects(classpath, args.all)
+  }
+
+  private def deserializeClassesOrObjects(
+      classpath: List[Path],
+      args: Seq[String]
+  ): Unit = {
     Scope { implicit scope =>
       val classLoader =
         ClassLoader.fromDisk {
           Config.empty.withClassPath(classpath)
         }
 
       for {
-        fileName <- args.all
+        className <- args
       } {
-        classLoader.load(Global.Top(fileName)) match {
+        classLoader.load(Global.Top(className)) match {
           case Some(defns) =>
             defns
               .sortBy(_.name.mangle)
               .foreach { d =>
                 println(d.show)
                 println()
               }
-          case None => fail(s"Not found class/object with name `${fileName}`")
+          case None => fail(s"Not found class/object with name `${className}`")
+        }
+      }
+    }
+  }
+
+  private def deserializeFiles(
+      classpath: List[Path],
+      args: Seq[String]
+  ): Unit = {
+
+    case class VirtDirFile(file: Path, virtDir: VirtualDirectory)
+
+    // Paths obtained from VirtualDirectory.files() are decoded as absolute,
```

Comment:
This is only true to JARs. and that's because JAR on itself is interpreted as a separate file system with it's own root directory etc